core/generated-sources/src/test/kotlin/org/jetbrains/kotlinx/dataframe/api/parse.kt

```diff
@@ -142,9 +157,340 @@ class ParseTests {
         columnOf("2022-01-23T04:29:40").parse().type shouldBe typeOf<LocalDateTime>()
     }
 
+    @Test
+    fun `can parse instants`() {
```

Comment:
can we ask stdlib for a non-throwing parsing function?

Reply:
ah, my bad, it's from Java, right?

Reply:
in any case, i want to understand better what copy-pasted parts are needed

Reply:
for kotlin Instants, we've got `parseOrNull`, a non-throwing parsing function :) For Java I managed to make my own `parseOrNull`.

I copied over a lot of tests from kotlinx-datetime such that we can catch functional changes when we update java/kotlinx-datetime versions.

The biggest thing I copied was regarding Kotlin Duration. It would be great if we could have a non-throwing `Duration.parseOrNull`. I solved this by creating a function `Duration.canParse` which contains copied logic but returns `false` instead of throwing an exception. For this I also put plenty of tests in place to catch functional changes if we bump Kotlin versoin.